client_ext_test.go

```diff
@@ -132,15 +132,6 @@ func TestClientPeer(t *testing.T) {
 		t.Parallel()
 		run(t,
 			connect.WithHTTPGet(),
-			connect.WithHTTPGetMaxURLSize(256, false),
-			connect.WithSendGzip(),
-		)
-	})
-	t.Run("connect+get fallback", func(t *testing.T) {
-		t.Parallel()
-		run(t,
-			connect.WithHTTPGet(),
-			connect.WithHTTPGetMaxURLSize(1, true),
```

Comment:
Unfortunately, exporting this out into an internal test is infeasible: as far as I can tell, there's no reasonable way to import the generated code from a given protobuf inside of the connect-go package without causing an import cycle, and moving the test to outside of the connect-go package would again prevent us from accessing the newly unexported option.

Reply:
We ought to be able to avoid cycles by skipping the code generated by protoc-gen-connect-go and using NewHandler and NewClient directly. This is one of Connect's benefits vs grpc-go - generics make generated code a convenience, not a necessity.

Reply:
911951962ab73a5731fdd802144e68d4dd3108a6